pkg/util/file_indexer.go

```diff
@@ -83,6 +85,60 @@ func resolveFilePath(directory string) (string, error) {
 	return directory, nil
 }
 
+// gitignoreFilePath gives the filepath of the .gitignore file in the context
+func gitIgnoreFilePath(directory string) (string, error) {
+	_, err := os.Stat(filepath.Join(directory))
+	if err != nil {
+		return "", err
+	}
+	return filepath.Join(directory, ".gitignore"), nil
+}
+
+// addOdoFileIndex adds odo-file-index.json to .gitignore
+func addOdoFileIndex(ignoreFile string) error {
+	var data []byte
+	file, err := os.OpenFile(ignoreFile, os.O_APPEND|os.O_RDWR, 0600)
+	if err != nil {
+		return errors.Wrap(err, "failed to open .gitignore file")
+	}
+	defer file.Close()
+
+	if data, err = ioutil.ReadFile(ignoreFile); err != nil {
+		return errors.Wrap(err, "failed reading data from .gitignore file")
+	}
+	// check whether .odo/odo-file-index.json is already in the .gitignore file
+	if !strings.Contains(string(data), ".odo/odo-file-index.json") {
+		if _, err := file.WriteString("\n" + ".odo/odo-file-index.json"); err != nil {
+			return errors.Wrap(err, "failed to Add odo-file-index.json to .gitignore file")
+		}
+	}
+	return nil
+}
+
+// Check .gitignore file exists or not
+func checkGitIgnoreFile(directory string) error {
+
+	ignoreFile, err := gitIgnoreFilePath(directory)
+	_, err = os.Stat(ignoreFile)
```

Comment:
Since we're doing `_, err = os.Stat(ignoreFile)`, I'm assuming that we're not interested in the `FileInfo` returned by `os.Stat`. In that case, why do we have a call to it here? I'm confused. Or am I looking at it the wrong way?

We're also not doing anything with the `err` here.

Reply:
Yes here we are just checking the existence of ```.gitignore``` path and if it does not exists the we are using ```err ``` to verify and create .gitignore if doesn't.

Reply:
the error checking should immediately follow `Stat` call. You should either move `if err!=nil` right after this, or move `os.Stat` down to where you are checking the error.